outport/process/outportDataProvider.go

```diff
@@ -150,41 +175,201 @@ func (odp *outportDataProvider) getSignersIndexes(header data.HeaderHandler) ([]
 	if odp.shardID == core.MetachainShardId {
 		return odp.createPoolForMeta(rewardsTxs)
 	}
 
 	return odp.createPoolForShard()
 }
 
-func (odp *outportDataProvider) createPoolForShard() *outportcore.Pool {
-	return &outportcore.Pool{
-		Txs:      WrapTxsMap(odp.txCoordinator.GetAllCurrentUsedTxs(block.TxBlock)),
-		Scrs:     WrapTxsMap(odp.txCoordinator.GetAllCurrentUsedTxs(block.SmartContractResultBlock)),
-		Rewards:  WrapTxsMap(odp.txCoordinator.GetAllCurrentUsedTxs(block.RewardsBlock)),
-		Invalid:  WrapTxsMap(odp.txCoordinator.GetAllCurrentUsedTxs(block.InvalidBlock)),
-		Receipts: WrapTxsMap(odp.txCoordinator.GetAllCurrentUsedTxs(block.ReceiptBlock)),
-		Logs:     odp.txCoordinator.GetAllCurrentLogs(),
+func (odp *outportDataProvider) createPoolForShard() (*outportcore.TransactionPool, error) {
+	txs, err := getTxs(odp.txCoordinator.GetAllCurrentUsedTxs(block.TxBlock))
+	if err != nil {
+		return nil, err
+	}
+
+	scrs, err := getScrs(odp.txCoordinator.GetAllCurrentUsedTxs(block.SmartContractResultBlock))
+	if err != nil {
+		return nil, err
+	}
+
+	rewards, err := getRewards(odp.txCoordinator.GetAllCurrentUsedTxs(block.RewardsBlock))
+	if err != nil {
+		return nil, err
+	}
+
+	invalidTxs, err := getTxs(odp.txCoordinator.GetAllCurrentUsedTxs(block.InvalidBlock))
+	if err != nil {
+		return nil, err
+	}
+
+	receipts, err := getReceipts(odp.txCoordinator.GetAllCurrentUsedTxs(block.ReceiptBlock))
+	if err != nil {
+		return nil, err
+	}
+
+	logs, err := getLogs(odp.txCoordinator.GetAllCurrentLogs())
+	if err != nil {
+		return nil, err
+	}
+
+	return &outportcore.TransactionPool{
+		Transactions:         txs,
+		SmartContractResults: scrs,
+		Rewards:              rewards,
+		InvalidTxs:           invalidTxs,
+		Receipts:             receipts,
+		Logs:                 logs,
+	}, nil
+}
+
+func (odp *outportDataProvider) createPoolForMeta(rewardsTxs map[string]data.TransactionHandler) (*outportcore.TransactionPool, error) {
+	txs, err := getTxs(odp.txCoordinator.GetAllCurrentUsedTxs(block.TxBlock))
+	if err != nil {
+		return nil, err
+	}
+
+	scrs, err := getScrs(odp.txCoordinator.GetAllCurrentUsedTxs(block.SmartContractResultBlock))
+	if err != nil {
+		return nil, err
+	}
+
+	rewards, err := getRewards(rewardsTxs)
+	if err != nil {
+		return nil, err
+	}
+
+	logs, err := getLogs(odp.txCoordinator.GetAllCurrentLogs())
+	if err != nil {
+		return nil, err
+	}
+
+	return &outportcore.TransactionPool{
+		Transactions:         txs,
+		SmartContractResults: scrs,
+		Rewards:              rewards,
+		Logs:                 logs,
+	}, nil
+}
+
+func getTxs(txs map[string]data.TransactionHandler) (map[string]*outportcore.TxInfo, error) {
+	ret := make(map[string]*outportcore.TxInfo, len(txs))
+
+	for txHash, txHandler := range txs {
+		tx, castOk := txHandler.(*transaction.Transaction)
+		txHashHex := getHexEncodedHash(txHash)
+		if !castOk {
+			return nil, fmt.Errorf("%w, hash: %s", errCannotCastTransaction, txHashHex)
```

Comment:
usually we used error naming like `WrongTypeAssertion` for these cases, not with cast, maybe use common error from errors package with custom transaction mesage, same for the others; not very strong opinion though

Reply:
There is a mixed practice here, we also use errors like `errCannotCastToBlockBody`/`ErrCannotCastUserAccountHandlerToVmCommonUserAccountHandler`

Not a strong opinion either, would leave it as it is